core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/NamedOperationResolver.java

```diff
@@ -122,42 +95,64 @@ public <T> T onFailure(final T result, final OperationChain<?> opChain, final Co
+     *
+     * @param operation {@link NamedOperation} or {@link OperationChain} to act on.
+     * @param user      User for the cache access.
+     * @param depth     Current recursive depth, will use limit set in class to
+     *                  continue or not.
+     * @return A list of resolved operations essentially flattened into plain
+     *         Operations and OperationChains.
+     */
+    private Collection<Operation> resolveNamedOperations(final Operation operation, final User user, final int depth) {
+        final Collection<Operation> updatedOperations = new ArrayList<>();
+        LOGGER.debug("Current resolver depth is: {}", depth);
+
+        // If a named operation resolve the operations within it
+        if (operation instanceof NamedOperation) {
+            NamedOperation<?, ?> namedOperation = (NamedOperation<?, ?>) operation;
+            LOGGER.debug("Resolving named operation called: {}", namedOperation.getOperationName());
+            try {
+                // Get the chain for the named operation from the cache
+                final OperationChain<?> namedOperationChain = cache
+                        .getNamedOperation(namedOperation.getOperationName(), user)
+                        .getOperationChain(namedOperation.getParameters());
+                // Update the operation inputs and add operation chain to the updated list
+                OperationHandlerUtil.updateOperationInput(namedOperationChain, namedOperation.getInput());
+
+                // Run again to resolve any nested operations in the chain before adding
+                namedOperationChain.updateOperations(resolveNamedOperations(namedOperationChain, user, depth + 1));
+                updatedOperations.add(namedOperationChain);
+
+            } catch (final CacheOperationException e) {
+                LOGGER.error("Exception resolving NamedOperation within the cache: {}", e.getMessage());
+                return Collections.singletonList(namedOperation);
             }
-        });
-
-        operations.updateOperations((Collection) updatedOperations);
-    }
+        // If given an operationChain resolve Operations inside it
+        } else if (operation instanceof OperationChain) {
+            LOGGER.debug("Resolving Operations in Operation Chain: {}", ((OperationChain<?>) operation).getOperations());
+            for (final Operation op : ((OperationChain<?>) operation).getOperations()) {
+                // Resolve if haven't hit the depth limit for named operations yet
+                // Note only need to check the depth here as when checking for a nested named operation it will always be
+                // recursively passed as an operation chain
+                if (depth < depthLimit) {
+                    updatedOperations.addAll(resolveNamedOperations(op, user, depth));
+                } else {
+                    LOGGER.warn("Nested depth limit hit resolving NamedOperations, operation maybe unresolved {}", op);
```

Comment:
So this information is not given to the user?
Will what happens next be:

1.  that the operation chain is then executed
2. `NamedOperationHandler` will error to the user and not explain its depth resolving issue?

I don't think I have seen any testing for this situation.

Reply:
In the event of ProxyStore or FederatedStore would this unresolved operation get sent to sub-graphs?
could this result in unexpected behaviour?

Reply:
The `Operation` will be left untouched in the chain, there's a chance the `Operation` checked here might not be a `NamedOperation` at which point this is not a problem, an additional check could probably be added here that checks if it is a `NamedOperation` and errors accordingly.

Reply:
This has been tweaked now to throw exception if limit is hit and there are still unresolved `NamedOperations` in the chain.

Reply:
Now that it errors, nothing should get to FedStore or ProxyStore.